packages/protocol/contracts/governance/Governance.sol

```diff
@@ -615,29 +629,42 @@ contract Governance is
    * @notice Approves the hash of a hotfix transaction(s).
    * @param hash The abi encoded keccak256 hash of the hotfix transaction(s) to be approved.
    */
-  function approveHotfix(bytes32 hash) external hotfixNotExecuted(hash) onlyApprover {
-    hotfixes[hash].approved = true;
-    emit HotfixApproved(hash);
+  function approveHotfix(
+    bytes32 hash
+  ) external hotfixNotExecuted(hash) onlyApproverOrSecurityCouncil {
+    if (msg.sender == approver) {
+      hotfixes[hash].approved = true;
+    } else {
+      if (isL2()) {
+        hotfixes[hash].counsilApproved = true;
+      } else {
+        revert("Hotfix approval by security council is not available on L1.");
+      }
+    }
+    emit HotfixApproved(hash, msg.sender);
   }
 
   /**
    * @notice Whitelists the hash of a hotfix transaction(s).
    * @param hash The abi encoded keccak256 hash of the hotfix transaction(s) to be whitelisted.
    */
-  function whitelistHotfix(bytes32 hash) external hotfixNotExecuted(hash) {
-    hotfixes[hash].whitelisted[msg.sender] = true;
+  function whitelistHotfix(bytes32 hash) external hotfixNotExecuted(hash) onlyL1 {
+    hotfixes[hash].deprecated_whitelisted[msg.sender] = true;
     emit HotfixWhitelisted(hash, msg.sender);
   }
 
   /**
    * @notice Gives hotfix a prepared epoch for execution.
    * @param hash The hash of the hotfix to be prepared.
    */
-  function prepareHotfix(bytes32 hash) external hotfixNotExecuted(hash) {
+  function prepareHotfix(bytes32 hash) external hotfixNotExecuted(hash) onlyL1 {
```

Comment:
Should add a way of invalidating the L2 hotfix if too much time has elapsed between approval and execution.

Reply:
prepareHotfix will reset the hotfix record if the `executionTimeLimit` has already been reached. This will require an additional round of approval by the approvers and Security Council, as well as an additional `prepareHotfix` step.